pacehrh/R/pace_cadre_process.R

```diff
@@ -0,0 +1,282 @@
+  tasks$NumContactsPerUnit[is.na(tasks$NumContactsPerUnit)] <- 0
+  tasks$NumContactsPer <-
+    tasks$NumContactsPerUnit + tasks$NumContactsAnnual
+
+  # Filter the Tasks table column set
+  tasks <-
+    tasks[, .(Indicator,
+              CommonName,
+              ClinicalOrNon,
+              ClinicalCat,
+              ServiceCat,
+              NumContactsPer)]
+
+  # Join Scenario information to the output
+  out <- dplyr::left_join(out,
+                          BVE$scenario[c("UniqueID",
+                                         "WeeksPerYr",
+                                         "HrsPerWeek",
+                                         "BaselinePop",
+                                         "DeliveryModel")],
+                          by = dplyr::join_by(Scenario_ID == UniqueID))
+
+  # Join Task information to the output
+  out <- dplyr::left_join(out,
+                          tasks,
+                          by = dplyr::join_by(Task_ID == Indicator))
+
+  # Create a stochastic hours per week value that's different for each year.
+  sdvalue <-
+    BVE$stochasticParams$p[BVE$stochasticParams$Value == "Hours per week"]
+
+  # TODO: find a more efficient calculation.
+  stochasticHrs <- out[, .(Count = .N), by = .(Trial_num, Year)]
+
+  stochasticHrs$HPW_stochastic <-
+    runif(
+      nrow(stochasticHrs),
+      min = BVE$scenario$HrsPerWeek * (1 - sdvalue),
+      max = BVE$scenario$HrsPerWeek * (1 + sdvalue)
+    )
+
+  stochasticHrs$HPW_stochastic[stochasticHrs$HPW_stochastic > 40] <- 40
+  stochasticHrs$Count <- NULL
+
+  out <- dplyr::left_join(out,
+                          stochasticHrs,
+                          by = dplyr::join_by(Trial_num == Trial_num, Year == Year))
+
+  if (!is.null(filepath)) {
+    data.table::fwrite(out,
+                       file = filepath,
+                       row.names = FALSE,
+                       na = "NA")
+  }
+
+  return(out)
+}
+
+
+#' Compute And Save Cadre Allocations
+#'
+#' Different healthcare delivery models involve different mixtures of types of
+#' healthcare workers ("Cadres"). `SaveCadreAllocations()` uses cadre
+#' information read earlier as part of running an experiment, then combines the
+#' allocation percentages with computed task times to produce a table showing
+#' how much time each cadre spends per year on each healthcare task.
+#'
+#' @param suiteResults Extended suite results as returned by `SaveExtendedSuiteResults()`
+#' @param filepath File location to save cadre allocations. Default = NULL
+#' @param annual Report as annual times instead of monthly? Default = TRUE
+#'
+#' @return data.table
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' }
+SaveCadreAllocations <- function(suiteResults, filepath = NULL, annual = TRUE) {
+  scenarioName <- BVE$scenario$UniqueID
+
+  # Do some sanity checking
+  if (!identical(GPE$years, unique(suiteResults$Year))) {
+    traceMessage("Configured year range (GPE) does not match reported year range")
```

Comment:
I ran into issue when loading suiteResults from a file where "Year" was read as integer and GPE$year is double, please cast them both to integer for identical comparison

Reply:
> @celiot-IDM Hi Charles, can you point me to where we currently store the annual overheads for all cadre members?

@rhanIDM I was going to leave you to guess ... just kidding :-)

```
# Read overhead times and save to file
cadreOverheadTimes <- pacehrh::SaveCadreOverheadData(filepath = "_COD.csv")

# Read overhead times without saving to file
cadreOverheadTimes <- pacehrh::SaveCadreOverheadData()
```

I went with the practices from the original post-processing, so times are in HOURS, not minutes as they are reported elsewhere. We should decided if we want to move to hours everywhere.